gensim/models/doc2vec.py

```diff
@@ -794,6 +794,24 @@ def train(self, documents=None, corpus_file=None, total_examples=None, total_wor
 
         """
         kwargs = {}
+
+        # Check if both documents and corpus_file are None
+        if corpus_file is None and documents is None:
+            raise TypeError("Either one of corpus_file or documents value must be provided.")
+
+        # Check if both documents and corpus_file are not None
+        if corpus_file is not None and documents is not None:
+            raise TypeError("Instead provide value to either of corpus_file or documents parameter but not both.")
+
+        # Check if documents is not None and iterable but not string type
+        if documents is not None and isinstance(documents, Iterable) and isinstance(documents, string_types):
+            raise TypeError("Documents must be an iterable of list and not a string type.")
+
+        # Check the type of corpus_file
+        if corpus_file is not None and not isinstance(corpus_file, string_types):
+            raise TypeError("""Parameter corpus_file of train() must be a
+                             string (path to an existing file) got %s instead.""" % corpus_file)
```

Comment:
Since `None` will already fail the is-`string_types` test, the check `is not None` is redundant.

Reply:
The use of a multiline string is poor here. Your exception message contains two lines, and a lot of unnecessary spaces.

I think you also don't need to mention the function name: it'll be obvious from the stack trace.

Reply:
@mpenkov noob question: should I resolve the conversations or the person who reviews the PR does it ?

Reply:
I think it makes sense for the original reviewer to do it, once they're happy their comments have been addressed.